### PR TITLE
fix: handle db connection error

### LINKED ISSUE #260: no function clause matching in EventStore.Storage.Appender.handle_response/1
It seems that connection errors causes some issues

```
** (FunctionClauseError) no function clause matching in EventStore.Storage.Appender.handle_response/1
    (eventstore 1.4.1) lib/event_store/storage/appender.ex:155: EventStore.Storage.Appender.handle_response({:error, %DBConnection.ConnectionError{
    message: "
    connection not available and request was dropped from queue after 110ms. This means requests are coming in and your connection pool cannot serve them fast enough. You can address this by:

    1. Ensuring your database is available and that you can connect to it
    2. Tracking down slow queries and making sure they are running fast enough
    3. Increasing the pool_size (although this increases resource consumption)
    4. Allowing requests to wait longer by increasing :queue_target and :queue_interval

    See DBConnection.start_link/2 for more information
    ",
    severity: :error,
    reason: :queue_timeout
}})

(eventstore 1.4.1) lib/event_store/storage/appender.ex:27: anonymous fn/5 in EventStore.Storage.Appender.append/4
  (elixir 1.14.0) lib/enum.ex:980: anonymous fn/3 in Enum.each/2
      (elixir 1.14.0) lib/enum.ex:4307: anonymous fn/3 in Enum.each/2
          (elixir 1.14.0) lib/stream.ex:1799: anonymous fn/3 in Enumerable.Stream.reduce/3
              (elixir 1.14.0) lib/stream.ex:288: Stream.after_chunk_while/2
                  (elixir 1.14.0) lib/stream.ex:1828: Enumerable.Stream.do_done/2
                      (elixir 1.14.0) lib/enum.ex:4307: Enum.each/2
                      "mfa=Commanded.Aggregates.Aggregate.execute_command/2

```

### PR DESCRIPTION
**Pull request by yordis**
closes #260

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -1,2 +1,2 @@
 elixir 1.14.1-otp-24
-erlang 24.3.4.6
+erlang 24.3.4.2
```
Comment (yordis) [inline .tool-versions:2]: the version `erlang 24.3.4.6` didn't exists so I had to fix this

File: test/storage/append_events_test.exs [hunk 2]
```diff
@@ -4,6 +4,13 @@ defmodule EventStore.Storage.AppendEventsTest do
   alias EventStore.{EventFactory, RecordedEvent, UUID}
   alias EventStore.Storage.{Appender, CreateStream}
 
+  test "append single event with a db connection error" do
```
Comment (yordis) [inline test/storage/append_events_test.exs:7]: Not working, I need some help trying to test this thou, I am not sure what could I do to trigger the new branch

**Comment by slashdotdash** [comment]
Closing this PR. It can be reopened with it is ready for merging,

**Comment by yordis** [comment]
Could you help me with the situation? I left it open `in draft` for visibility and hopefully get some help since you will remand a test case for it.
I would appreciate if you can help with a bit here

File: test/storage/append_events_test.exs (hunk 2 above)
Comment (slashdotdash) [reply test/storage/append_events_test.exs:7]: Test fixed in b0a5e022d2d45af9dcea08ffef721a66569c4ca8.

**Comment by slashdotdash** [comment]
Finished up the changes and merged into master branch in 07bb96f17d391ef3e264bee90e0a51422bf1a56f.

**Pull request merged**